src/components/MetadataDisplay/MetadataDisplay.js

```diff
@@ -1,57 +1,156 @@
+  const [showCanvasMetadata, setShowCanvasMetadata] = React.useState();
 
+  let canvasesMetadataRef = React.useRef();
+  const setCanvasesMetadata = (m) => {
+    _setCanvasesMetadata(m);
+    canvasesMetadataRef.current = m;
+  };
+  /**
+   * On the initialization of the component read metadata from the Manifest
+   * and/or Canvases based on the input props and set the initial set(s) of
+   * metadata in the component's state
+   */
   React.useEffect(() => {
     if (manifest) {
-      let parsedMetadata = parseMetadata(manifest);
-      if (!displayTitle) {
-        parsedMetadata = parsedMetadata.filter(md => md.label.toLowerCase() != 'title');
+      // Display Canvas metadata only when specified in the props
+      const showCanvas = displayOnlyCanvasMetadata || displayAllMetadata;
+      setShowCanvasMetadata(showCanvas);
+      const showManifest = !displayOnlyCanvasMetadata || displayAllMetadata;
+      setShowManifestMetadata(showManifest);
+
+      // Parse metadata from Manifest
+      const parsedMetadata = getMetadata(manifest, showCanvas);
+
+      // Set Manifest and Canvas metadata in the state variables according to props
+      if (showCanvas) {
+        setCanvasesMetadata(parsedMetadata.canvasMetadata);
+        setCanvasMetadataInState();
+      }
+      if (showManifest) {
+        let manifestMeta = parsedMetadata.manifestMetadata;
+        if (!displayTitle) {
+          manifestMeta = manifestMeta.filter(md => md.label.toLowerCase() != 'title');
+        }
+        setManifestMetadata(manifestMeta);
       }
-      setMetadata(parsedMetadata);
     }
   }, [manifest]);
 
-  if (metadata && metadata.length > 0) {
-    return (
-      <div
-        data-testid="metadata-display"
-        className="ramp--metadata-display">
-        {showHeading && (
-          <div className="ramp--metadata-display-title" data-testid="metadata-display-title">
-            <h4>Details</h4>
-          </div>
-        )}
-        <div className="ramp--metadata-display-content">
-          {metadata.map((md, i) => {
-            return (
-              <React.Fragment key={i}>
-                <dt>{md.label}</dt>
-                <dd dangerouslySetInnerHTML={{ __html: md.value }}></dd>
-              </React.Fragment>
-            );
-          })
-          }
-        </div>
-      </div>
-    );
-  } else {
-    return (<div
+  /**
+   * When displaying current Canvas's metadata in the component, update the metadata
+   * in the component's state listening to the canvasIndex changes in the central
+   * state
+   */
+  React.useEffect(() => {
+    if (canvasIndex >= 0 && showCanvasMetadata) {
+      setCanvasMetadataInState();
+    }
+  }, [canvasIndex]);
+
+  /**
+   * Set canvas metadata in state
+   */
+  const setCanvasMetadataInState = () => {
+    let canvasData = canvasesMetadataRef.current
+      .filter((m) => m.canvasindex === canvasIndex)[0].metadata;
+    if (!displayTitle && displayOnlyCanvasMetadata) {
```

Comment:
I don't think there's a problem with removing the title from the display when `displayTitle=false` for both Manifest and Canvas metadata. 

My thoughts were that with `displayTitle=false` it would always apply to the first set of metadata in the display because that's the main resource displayed in Ramp and when both Manifest and Canvas metadata is displayed the main resource would be the Manifest not the Canvas.

I'm okay with adding this filter for both Canvas and Manifest metadata when metadata for both are displayed. I'll make that change.